src/EditorFeatures/Core/FindUsages/FindUsagesHelpers.cs

```diff
@@ -47,95 +47,111 @@ public static string GetDisplayName(ISymbol symbol)
 
             var mapping = await mappingService.MapSymbolAsync(document, symbol, cancellationToken).ConfigureAwait(false);
             if (mapping == null)
-            {
                 return null;
-            }
 
             return (mapping.Symbol, mapping.Project);
         }
 
-        public static async Task<(ISymbol symbol, Project project, ImmutableArray<ISymbol> implementations, string message)?> FindImplementationsAsync(Document document, int position, CancellationToken cancellationToken)
+        public static async Task<(Solution solution, SymbolAndProjectId symboAndProjectId, ImmutableArray<SymbolAndProjectId> implementations, string message)?> FindSourceImplementationsAsync(Document document, int position, CancellationToken cancellationToken)
         {
-            var symbolAndProject = await GetRelevantSymbolAndProjectAtPositionAsync(
+            var symbolAndProjectOpt = await GetRelevantSymbolAndProjectAtPositionAsync(
                 document, position, cancellationToken).ConfigureAwait(false);
-            if (symbolAndProject == null)
-            {
+            if (symbolAndProjectOpt == null)
                 return null;
-            }
 
-            return await FindImplementationsAsync(
-                symbolAndProject?.symbol, symbolAndProject?.project, cancellationToken).ConfigureAwait(false);
+            var (symbol, project) = symbolAndProjectOpt.Value;
+            var symbolAndProjectId = new SymbolAndProjectId(symbol, project.Id);
+            return await FindSourceImplementationsAsync(
+                symbolAndProjectId, project.Solution, cancellationToken).ConfigureAwait(false);
         }
 
-        private static async Task<(ISymbol symbol, Project project, ImmutableArray<ISymbol> implementations, string message)?> FindImplementationsAsync(
-            ISymbol symbol, Project project, CancellationToken cancellationToken)
+        private static async Task<(Solution solution, SymbolAndProjectId symbolAndProjectId, ImmutableArray<SymbolAndProjectId> implementations, string message)?> FindSourceImplementationsAsync(
+            SymbolAndProjectId symbolAndProjectId, Solution solution, CancellationToken cancellationToken)
```

Comment:
💡 Put `solution` as the first argument for consistency with other methods and the ordering of tuple elements in return values?

Reply:
Yup.  Happy to do that (in followup)! :)